engine/src/main/java/org/pentaho/di/trans/steps/rest/StringMessageBodyWriter.java

```diff
@@ -0,0 +1,69 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2020 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.rest;
+
+import com.sun.jersey.core.impl.provider.entity.StringProvider;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+
+/**
+ * This class is a custom provider to replace the default com.sun.jersey.core.impl.provider.entity.StringProvider
+ * which it has a bad implementation related with get size.
+ *
+ *
+ */
+@Produces( { "text/plain", "*/*" } )
+public final class StringMessageBodyWriter implements MessageBodyWriter<String> {
+
+  private StringProvider stringProvider;
```

Comment:
can be final as well

Reply:
It cannot be final to this class be able to be mocked.

Reply:
Now that the UTs were fixed, this can be final.